cli/src/pcluster/launch_template_utils.py

```diff
@@ -0,0 +1,46 @@
+from pcluster.constants import OS_MAPPING
+
+
+def get_block_device_mappings_template(create_block_device_mapping_for_virt, create_block_device_mapping_for_ebs):
```

Comment:
I don't have strong feelings on this, functional implementation (2nd commit) was because I had later read in Effective Java 3rd edition the suggestion to use functional interfaces and composition instead of inheritance like in the traditional GoF design pattern (I realize that this is Python, but the general point being made seemed relevant since it was in the context of the introduction of functional interfaces, and Python supports higher order functions). Will proceed to revert that commit and make these methods of a single class.